custom_components/animated_scenes/animations.py

```
@@ -1142,50 +1101,82 @@ async def add_lights_to_animation(self, data: dict[str, Any]) -> None:
+            tracked_animations = list(self._light_animations.get(light, []))
+            owner = self.light_owner.get(light)
+            if owner is not None and owner not in tracked_animations:
+                tracked_animations.append(owner)
+            for animation in tracked_animations:
+                _LOGGER.info("Removing light '%s' from animation '%s'", light, animation.name)
                 affected_animations.add(animation)
                 animation.remove_light(light)
-                updates.append(self.release_light(animation, light, True, skip_restore))
+            self._light_animations.pop(light, None)
+            if owner is not None:
+                updates.append(self.release_light(owner, light, True, skip_restore))
```

Ruff (0.16.0)

[warning] Boolean positional value in function call

(FBT003)